acvm-repo/acvm/tests/solver.rs

```diff
@@ -722,3 +724,531 @@ fn memory_operations() {
+        prop_assert_eq!(results, expected_results)
+    }
+
+    #[test]
+    fn bigint_mul_zero_r((xs, modulus) in bigint_with_modulus()) {
+        let zero = bigint_zeroed(&xs);
+        let expected_results = drop_use_constant(&zero);
+        let results = bigint_solve_binary_op(bigint_mul_op(), modulus, xs, zero);
+        prop_assert_eq!(results, expected_results)
+    }
+
+    // TODO(https://github.com/noir-lang/noir/issues/5579): Fails on 49, see bigint_add_zero_l_single_case_49
+    #[test]
+    #[should_panic(expected = "Test failed: assertion failed: `(left == right)`")]
+    fn bigint_mul_one_l((xs, modulus) in bigint_with_modulus()) {
+        let one = bigint_to_one(&xs);
+        let expected_results: Vec<_> = drop_use_constant(&xs);
+        let results = bigint_solve_binary_op(bigint_mul_op(), modulus, one, xs);
+        prop_assert_eq!(results, expected_results)
+    }
+
+    // TODO(https://github.com/noir-lang/noir/issues/5579): Fails on 49, see bigint_add_zero_l_single_case_49
+    #[test]
+    #[should_panic(expected = "Test failed: assertion failed: `(left == right)`")]
+    fn bigint_mul_one_r((xs, modulus) in bigint_with_modulus()) {
+        let one = bigint_to_one(&xs);
+        let expected_results = drop_use_constant(&xs);
+        let results = bigint_solve_binary_op(bigint_mul_op(), modulus, xs, one);
+        prop_assert_eq!(results, expected_results)
+    }
+
+    #[test]
+    fn bigint_sub_self((xs, modulus) in bigint_with_modulus()) {
+        let expected_results = drop_use_constant(&bigint_zeroed(&xs));
+        let results = bigint_solve_binary_op(bigint_sub_op(), modulus, xs.clone(), xs);
+        prop_assert_eq!(results, expected_results)
+    }
+
+    // TODO(https://github.com/noir-lang/noir/issues/5579): Fails on 49, see bigint_add_zero_l_single_case_49
+    #[test]
+    #[should_panic(expected = "Test failed: assertion failed: `(left == right)`")]
+    fn bigint_sub_zero((xs, modulus) in bigint_with_modulus()) {
+        let zero = bigint_zeroed(&xs);
+        let expected_results: Vec<_> = drop_use_constant(&xs);
+        let results = bigint_solve_binary_op(bigint_sub_op(), modulus, xs, zero);
+        prop_assert_eq!(results, expected_results)
+    }
+
+    #[test]
+    fn bigint_sub_one((xs, modulus) in bigint_with_modulus()) {
+        let one = bigint_to_one(&xs);
+        let expected_results: Vec<_> = drop_use_constant(&xs);
+        let results = bigint_solve_binary_op(bigint_sub_op(), modulus, xs, one);
+        prop_assert!(results != expected_results, "{:?} == {:?}", results, expected_results)
+    }
+
+    #[test]
+    fn bigint_div_self((xs, modulus) in bigint_with_modulus()) {
+        let one = drop_use_constant(&bigint_to_one(&xs));
+        let results = bigint_solve_binary_op(bigint_div_op(), modulus, xs.clone(), xs);
+        prop_assert_eq!(results, one)
+    }
+
+    #[test]
+    fn bigint_div_by_zero((xs, modulus) in bigint_with_modulus()) {
```

Comment:
isn't supposed to fail?

Reply:
I'm not sure. In some languages, division by zero is well-defined, e.g. `x / 0 == 0` in Coq.

Reply:
I think we should be consistent with what we do with Field, i.e it should fail, rather than with other languages.

Reply:
Opened an issue to follow up: https://github.com/noir-lang/noir/issues/5645